lib/android-helpers.js

```diff
@@ -60,12 +60,20 @@ helpers.prepareEmulator = async function (adb, opts) {
       return;
     }
   }
+  avdArgs = this.prepareAVDArgs(opts, adb, avdArgs);
+  await adb.launchAVD(avd, avdArgs, language, locale, avdLaunchTimeout,
+                      avdReadyTimeout);
+};
+
+helpers.prepareAVDArgs = function (opts, adb, avdArgs) {
   if (!_.isUndefined(opts.networkSpeed)) {
     let networkSpeed = this.ensureNetworkSpeed(adb, opts.networkSpeed);
     avdArgs += ` -netspeed ${networkSpeed}`;
   }
-  await adb.launchAVD(avd, avdArgs, language, locale, avdLaunchTimeout,
-                      avdReadyTimeout);
+  if (!_.isUndefined(opts.isHeadless) && opts.isHeadless === true) {
+    avdArgs += ' -no-window';
```

Comment:
it looks a bit redundant to me that we add strings leading with spaces. I'd rather create an array of additional args and push values there and then do something like avdArgs + ` ${array.join(' ')}` at the end.